### PR TITLE
Dscs 93

### PR DESCRIPTION
**Pull request by snezanam**
@scoen, please have a look, it has changes to include copies for the collection for alias report and expiry report query.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
   <groupId>au.gov.nla</groupId>
   <artifactId>amberdb</artifactId>
-  <version>1.2.2-SNAPSHOT</version>
+  <version>1.2.3-SNAPSHOT</version>
```
Comment (AaronCMuller) [inline pom.xml]: This is not actually required. Snapshots are meant to be overridden.

File: src/amberdb/AmberSession.java [hunk 2]
```diff
@@ -867,6 +867,24 @@ private void increment(Map<String, Integer> countMap, String key) {
         return aliasMap;
     }
     
+    public List<Work> getExpiryReport(String expiryYear, String collection) {
+
+        ObjectsWithPropertyInCollectionQuery avq = new ObjectsWithPropertyInCollectionQuery(getAmberGraph());
+        List<Vertex> results = avq.getExpiryReport(expiryYear, collection);
+        List<Work> works = new ArrayList<Work>();
+        System.out.println("SIZE=" + results.size());
```
Comment (AaronCMuller) [inline src/amberdb/AmberSession.java]: Either make this a log#info statement, or remove please.

File: src/amberdb/query/ObjectsWithPropertyInCollectionQuery.java [hunk 3]
```diff
@@ -67,4 +65,54 @@ public ObjectsWithPropertyInCollectionQuery(AmberGraph graph) {
         }
         return vertices;
     }
+    
+    
+    public List<Vertex> getExpiryReport(String expiryYear, String collectionName) {
```
Comment (AaronCMuller) [inline src/amberdb/query/ObjectsWithPropertyInCollectionQuery.java]: @scoen Is it possible for a list of works to be returned here?

File: src/amberdb/AmberSession.java (hunk 2 above)
Comment (terenceingram) [inline src/amberdb/AmberSession.java]: I think remove the line.

File: src/amberdb/query/ObjectsWithPropertyInCollectionQuery.java [hunk 4]
```diff
@@ -67,4 +65,54 @@ public ObjectsWithPropertyInCollectionQuery(AmberGraph graph) {
...
+            q.bind("internalAccessConditions", "internalAccessConditions");
+            q.bind("closed", "Closed");
+            q.bind("typeName", "type");
+            q.bind("type1", AmberProperty.encode("Work"));
+            q.bind("collection", "collection");
+            q.bind("value", AmberProperty.encode(collectionName));
+            q.execute();
+            h.commit();
+
+            // and reap the rewards
```
Comment (AaronCMuller) [inline src/amberdb/query/ObjectsWithPropertyInCollectionQuery.java]: Please rewrite this comment to be less informal, and possibly more informative?

**Comment by AaronCMuller** [comment]
Other than my comments, the rest looks good.

**Comment by snezanam** [comment]
@scoen  @AaronCMuller please have another look, I have renamed the Query class and its methods and added a unit test. To generate works from vertices we need framed graph which is  available in amber session, in the query class we have AmberGraph @scoen can we easily do this?

File: src/amberdb/AmberSession.java [hunk 5]
```diff
@@ -867,6 +868,24 @@ private void increment(Map<String, Integer> countMap, String key) {
         return aliasMap;
     }
     
+    public List<Work> getExpiryReport(String expiryYear, String collection) {
+
+        ObjectsWithPropertyReportQuery avq = new ObjectsWithPropertyReportQuery(getAmberGraph());
+        String tableDrop = getAmberGraph().getTempTableDrop();
+        List<Vertex> results = avq.generateExpiryReport(expiryYear, collection, tableDrop);
```
Comment (scoen) [inline src/amberdb/AmberSession.java]: Would you mind doing the h2 check (tabledrop) inside the query itself ? Like in generateDuplicateAlias report, call graph.getTempTableDrop(). I'd prefer this code 'carbuncle' be isolated to the query class if possible. Sorry about needing to do that.

Comment (snezanam) [inline src/amberdb/AmberSession.java]: That sounds good.

**Pull request merged**